Modules/commandcontext.py

```diff
@@ -0,0 +1,104 @@
+from functools import reduce
+from operator import xor
+from typing import List, Union
+
+import pydle
+
+from Modules.user import User
+
+
+class CommandContext(object):
+    """
+    Object to hold information on the user who invoked a command as well as where and how the
+    command was invoked.
+    """
+
+    def __init__(self, bot: 'MechaClient', words: List[str], words_eol: List[str], user,
+                 target: str):
+        """
+        Initializes a new `CommandContext` object with the provided info.
+
+        Arguments:
+            user ():
+            bot (pydle.BasicClient): This bot's instance.
+            words ([str]): List of all the words of the message which invoked the command. Does not
+                include the command prefix char.
+            words_eol ([str]): Same as *words* above, but each element including the word and
+                everything up to the end of the message.
+        """
+        self._bot = bot
+        self._words = words
+        self._words_eol = words_eol
+        self._user = user
+        self._hash = None
+        self._target = target
+
+    bot: 'MechaClient' = property(lambda self: self._bot)
+    """MechaClient instance"""
+    words: [str] = property(lambda self: self._words)
+    words_eol: list = property(lambda self: self._words_eol)
+    """words deliminated to EOL"""
+    user: User = property(lambda self: self._user)
+    """invoking IRC user"""
+
+
+    @classmethod
+    async def from_bot_user(cls, bot: pydle.BasicClient, nickname: str, target: str,
```

Comment:
With the (very good) `User.from_bot` method existing, there might not be much use in this one, as it doesn't serve much use over the constructor. The `User` object could be (and would probably make more sense to be) created by the caller.